src/gax-internal/src/grpc.rs

```diff
@@ -67,48 +68,34 @@ impl Client {
         Request: prost::Message + 'static + Clone,
         Response: prost::Message + Default + 'static,
     {
+        let headers =
+            Self::make_headers(&self.credentials, api_client_header, request_params).await?;
```

Comment:
The headers also include the credentials, and we need those to be created inside the loop.  Can we (maybe) split this in two: create fixed headers outside the loop, and then append the credential headers inside the loop?

Reply:
> we need those to be created inside the loop.

Yes, thanks for catching. :facepalm: Ugh, I got lost in the sauce on this refactor.

#1839 adds a test that would have caught this sooner.

Reply:
Fixed